realm/realm-library/src/main/java/io/realm/TypeSelectorForMap.java

```diff
@@ -107,18 +111,20 @@ protected V putRealmModel(BaseRealm baseRealm, OsMap osMap, K key, @Nullable V v
     private <T> RealmResults<T> produceResults(BaseRealm baseRealm,
                                                Pair<Table, Long> tableAndValuesPtr,
                                                boolean forPrimitives,
-                                               @Nullable Class<T> clazz) {
-        if (baseRealm instanceof Realm) {
-            Realm realm = (Realm) baseRealm;
-            Long valuesPtr = tableAndValuesPtr.second;
-            OsResults osResults = OsResults.createFromMap(baseRealm.sharedRealm, valuesPtr);
-            if (clazz != null) {
-                return new RealmResults<>(realm, osResults, clazz, forPrimitives);
-            }
-            throw new IllegalStateException("MapValueOperator missing class.");
-        }
+                                               Class<T> clazz) {
+        Long valuesPtr = tableAndValuesPtr.second;
+        OsResults osResults = OsResults.createFromMap(baseRealm.sharedRealm, valuesPtr);
+        return new RealmResults<>(baseRealm, osResults, clazz, forPrimitives);
+    }
 
-        throw new UnsupportedOperationException("Add support for 'values' for DynamicRealms.");
+    @Override
+    Class<V> getValueClass() {
+        return valueClass;
+    }
+
+    @Override
+    String getValueClassName() {
+        return null;
```

Comment:
This implementation is only to be called when working with regular Realms, right?

Reply:
Yes. I am going to return the actual value class name.